pkg/scheduler/objects/application_state.go

```diff
@@ -106,6 +108,10 @@ func NewAppState() *fsm.FSM {
 				Name: FailApplication.String(),
 				Src:  []string{Failing.String()},
 				Dst:  Failed.String(),
+			}, {
+				Name: ResumeApplication.String(),
+				Src:  []string{New.String(), Accepted.String()},
+				Dst:  Resuming.String(),
```

Comment:
<img width="1118" alt="YuniKorn__Online_Whiteboard_for_Visual_Collaboration" src="https://user-images.githubusercontent.com/14214570/120569352-94ab5a80-c3ca-11eb-86ff-7ef94070ba7e.png">

in this chart, I think when placeholder times out, the app could be in any state among [Accepted, Starting, Running, Waiting, Completing]. But we only handle the states `New` and `Accepted`, will this be able to cover all other cases?

Reply:
In all other cases (if the app is Running, Starting or Completing) we don't fail the application. In this case we just delete the remaining placeholders and let the application finish. You can see this part here: https://github.com/apache/incubator-yunikorn-core/blob/master/pkg/scheduler/objects/application.go#L296-L312

Reply:
We don't have anymore Waiting state. It was changed to Completing, so I think the Soft style will have impact only for the other cases when we would fail the application.